calicoctl/calico_ctl/diags.py

```diff
@@ -144,23 +160,22 @@ def save_diags(log_dir, runtime="docker"):
             print "  - Error running ipset. Maybe you need to run as root."
             f.writelines("Error running ipset: %s\n" % e)
 
-    # Ask Felix to dump stats to its log file - ignore errors as the
-    # calico/node container might not be running.
-    subprocess.call(["pkill", "-SIGUSR1", "felix"])
-
     # If running under rkt, get the journal for the calico/node container.
     print "Copying journal for calico-node.service"
-    f = open(temp_diags_dir + "/calico-node.journal", "w")
-    try:
-        journal = subprocess.check_output(["journalctl", "-u", 
-                                           "calico-node.service", 
-                                           "--no-pager"])
-    except OSError:
-        print "Unable to copy journal"
-    else:
-        f.write(journal)
-    finally:
-        f.close()
+    with DiagsErrorWriter(temp_diags_dir, "calico-node.journal") as f:
+        try:
+            journalctl = sh.Command._create("journalctl")
+            f.write(journalctl("-u", "calico-node.service", "--no-pager"))
+        except sh.CommandNotFound as e:
+            print "  - Missing command: %s" % e.message
+            f.writelines("Missing command: %s\n" % e.message)
+        except sh.ErrorReturnCode_1 as e:
+            print "  - Error running journalctl."
+            f.writelines("Error running journalctl: %s\n" % e)
+
+    # Ask Felix to dump stats to its log file - ignore errors as the
```

Comment:
I think this bit about Felix needs to go before the journalctl dump.  When running under rkt, calico/node logs all go to journalctl.